### PR TITLE
Added an edge detection node to assist with focus so we don't have to keep guessing

### PR DESCRIPTION
**Pull request by ajernandes**

### CODE REVIEW

File: rktl_perception/launch/focus_assist.launch [hunk 1]
```diff
@@ -0,0 +1,6 @@
+<launch>
```
Comment (dgerblick) [inline rktl_perception/launch/focus_assist.launch:1]: This should be modified to start each camera. That way we don't have to run the full stack just to focus the cameras.

File: rktl_perception/src/focus_vis_node.cpp [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+    {}
+
+void FocusNode::FocusCallback(const sensor_msgs::ImageConstPtr& msg, const sensor_msgs::CameraInfoConstPtr& info) {
+    try {
+        /* Convert the ROS message into a cv_ptr & dereferencing the pointer to an OpenCV Mat */
+        cv_bridge::CvImagePtr cv_ptr = cv_bridge::toCvCopy(msg, "bgr8");
+        cv::Mat current_frame = cv_ptr->image;
+        cv::Mat frame_gray, frame_blurred, frame_edge;
+        
+        #if false /* shouldn't be necessary on the new computers */
```
Comment (dgerblick) [inline rktl_perception/src/focus_vis_node.cpp:26]: Why is this here? I think we can just remove it

File: rktl_perception/src/focus_vis.cpp [hunk 3]
```diff
@@ -0,0 +1,17 @@
+/* Node to detect the ball's position on the field.
```
Comment (dgerblick) [inline rktl_perception/src/focus_vis.cpp:1]: This is false

File: rktl_perception/src/focus_vis_node.cpp [hunk 4]
```diff
@@ -0,0 +1,51 @@
+/* Node to detect the ball's position on the field.
```
Comment (dgerblick) [inline rktl_perception/src/focus_vis_node.cpp:1]: This is also false

File: rktl_perception/include/rktl_perception/focus_vis_node.h [hunk 5]
```diff
@@ -0,0 +1,50 @@
+/* Node to detect the ball's position on the field.
```
Comment (dgerblick) [inline rktl_perception/include/rktl_perception/focus_vis_node.h:1]: This is very false

File: rktl_perception/include/rktl_perception/focus_vis_node.h [hunk 6]
```diff
@@ -0,0 +1,50 @@
...
+ * License:
+ *   BSD 3-Clause License
+ *   Copyright (c) 2020, Autonomous Robotics Club of Purdue (Purdue ARC)
+ *   All rights reserved.
+ */
+
+#pragma once
```
Comment (dgerblick) [inline rktl_perception/include/rktl_perception/focus_vis_node.h:8]: This is completely fine and works perfectly but I just don't like it and I'm leaving this comment out of pure spite.

**Review by dgerblick — CHANGES_REQUESTED** [review]
Made a few comments. Most you can ignore, The only one that matters is the launch file one.

File: rktl_perception/src/focus_vis_node.cpp (hunk 2 above)
Comment (ajernandes) [reply rktl_perception/src/focus_vis_node.cpp:26]: I presume you're talking about the resizing, which is gone. if you're talking about the `cv::Mat` stuff, that has to stay.

File: rktl_perception/src/focus_vis.cpp (hunk 3 above)
Comment (ajernandes) [reply rktl_perception/src/focus_vis.cpp:1]: mmhmm

File: rktl_perception/include/rktl_perception/focus_vis_node.h (hunk 6 above)
Comment (ajernandes) [reply rktl_perception/include/rktl_perception/focus_vis_node.h:8]: I'm leaving it for the same reason. idek what it does, James had it in code that I copied and pasted a year ago.

File: rktl_perception/include/rktl_perception/focus_vis_node.h (hunk 5 above)
Comment (ajernandes) [reply rktl_perception/include/rktl_perception/focus_vis_node.h:1]: mmhmm

File: rktl_perception/src/focus_vis_node.cpp (hunk 4 above)
Comment (ajernandes) [reply rktl_perception/src/focus_vis_node.cpp:1]: k.

File: rktl_perception/launch/focus_assist.launch (hunk 1 above)
Comment (ajernandes) [reply rktl_perception/launch/focus_assist.launch:1]: Good point.

File: rktl_perception/src/focus_vis_node.cpp (hunk 2 above)
Comment (dgerblick) [reply rktl_perception/src/focus_vis_node.cpp:26]: I meant the \#if false stuff

Comment (ajernandes) [reply rktl_perception/src/focus_vis_node.cpp:26]: gone & pushed

**Review by dgerblick — APPROVED** [review]
I rescind all previous comments (except for the pragma once stuff, I'm team include guards until I die)
